Author: Kevin Doren <kevin@doren.org>
Description: Revert JUCE commit 802f33b of Jul 27, 2021 (Avoid setting realtime priority).

Index: sonobus/deps/juce/modules/juce_core/native/juce_posix_SharedCode.h
===================================================================
--- sonobus.orig/deps/juce/modules/juce_core/native/juce_posix_SharedCode.orig.h
+++ sonobus/deps/juce/modules/juce_core/native/juce_posix_SharedCode.h
@@ -959,11 +959,9 @@
 
 bool Thread::setThreadPriority (void* handle, int priority)
 {
-    constexpr auto maxInputPriority = 10;
-    constexpr auto lowestRealtimePriority = 8;
-
     struct sched_param param;
     int policy;
+    priority = jlimit (0, 10, priority);
 
     if (handle == nullptr)
         handle = (void*) pthread_self();
@@ -971,19 +969,12 @@
     if (pthread_getschedparam ((pthread_t) handle, &policy, &param) != 0)
         return false;
 
-    policy = priority < lowestRealtimePriority ? SCHED_OTHER : SCHED_RR;
-
-    const auto minPriority = sched_get_priority_min (policy);
-    const auto maxPriority = sched_get_priority_max (policy);
-
-    param.sched_priority = [&]
-    {
-        if (policy == SCHED_OTHER)
-            return 0;
+    policy = priority == 0 ? SCHED_OTHER : SCHED_RR;
 
-        return jmap (priority, lowestRealtimePriority, maxInputPriority, minPriority, maxPriority);
-    }();
+    const int minPriority = sched_get_priority_min (policy);
+    const int maxPriority = sched_get_priority_max (policy);
 
+    param.sched_priority = ((maxPriority - minPriority) * priority) / 10 + minPriority;
     return pthread_setschedparam ((pthread_t) handle, policy, &param) == 0;
 }
 
